flycheck.el

```diff
@@ -3610,6 +3663,40 @@ list."
       ;; highlighting
       (flycheck-error-list-highlight-errors preserve-pos))))
 
+(defun flycheck-error-list-mode-line-filter-indicator ()
+  "Create a string representing the current error list filter."
+  (if flycheck-error-list-minimum-level
+      (format " [>= %s]" (symbol-name flycheck-error-list-minimum-level))
+    ""))
+
+(defun flycheck-error-list-set-filter (level)
+  "Restrict the error list to errors at level LEVEL or higher."
+  (interactive
+   (list (read-flycheck-error-level
+          "Minimum error level (errors at lower levels will be hidden): ")))
+  (-when-let (buf (get-buffer flycheck-error-list-buffer))
+    (with-current-buffer buf
+      (set (make-local-variable 'flycheck-error-list-minimum-level)
+           (and (flycheck-error-level-p level) level)))
+    (flycheck-error-list-refresh)
+    (flycheck-error-list-recenter-at (point-min))))
+
+(defun flycheck-error-list-reset-filter ()
+  "Remove filters and show all errors in the error list."
+  (interactive)
+  (flycheck-error-list-set-filter
+   (default-value 'flycheck-error-list-minimum-level)))
```

Comment:
I think we can just use `(kill-local-variable 'flycheck-error-list-minimum-level)` here, to remove the buffer-local binding, which makes Emacs fall back to the global value, i.e. the default value, automatically.